src/main/java/prefs/Preferences.java

```diff
@@ -10,160 +16,296 @@
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The filename of the config file on disk
+     * @param configClass The class of the config to load
+     * @return The loaded Config instance
      */
-    public static Preferences create(String configFileName) {
-        return new Preferences(configFileName, true);
-    }
+    private <T> T loadConfig(String configDirectory, String configFileName, Class<T> configClass) {
+        String fileContents = DEFAULT_FILE_CONTENTS;
+        if (FileHelper.fileExists(configDirectory, configFileName)) {
+            try {
+                fileContents = FileHelper.loadFileContents(configDirectory, configFileName);
+            } catch (IOException e) {
+                // if we can't read the file, just ignore the existing files
+                HTLog.error(logger, e);
+                logger.error(e.toString());
+                fileContents = DEFAULT_FILE_CONTENTS;
+            }
+        }
 
-    public void saveGlobalConfig() {
-        fileHandler.saveGlobalConfig(global);
+        return createConfig(fileContents, configClass);
     }
 
-    private void initGlobalConfig() {
-        global = fileHandler.initGlobalConfig();
+    /**
+     * Saves the session and user configs to file
+     */
+    private void save() {
+        try {
+            saveConfig(sessionConfig, configDirectory, sessionConfigFileName, SessionConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save session config");
+        }
+
+        try {
+            saveConfig(userConfig, configDirectory, userConfigFileName, UserConfig.class);
+
+        } catch (IOException e) {
+            HTLog.error(logger, e);
+            logger.error("Could not save user config");
+        }
     }
 
-    private void loadGlobalConfig() {
-        global = fileHandler.loadGlobalConfig();
+    /**
+     * Saves a Config to file
+     * @param config The config object to save.
+     * @param configDirectory The directory that the config file is held in.
+     * @param configFileName The file name of the config file on disk
+     * @param configClass The class of the config object
+     */
+    private <T> void saveConfig(T config,
+                            String configDirectory,
+                            String configFileName,
+                            Class configClass) throws IOException {
+        String jsonString = JsonHelper.toJsonString(config, configClass);
+        FileHelper.writeFileContents(configDirectory, configFileName, jsonString);
     }
 
-    // Last login credentials. While the main UI is running (i.e. logged in successfully), last login
-    // credentials are guaranteed to be the current user's credentials thanks to setLastLoginCredentials
-    // being called immediately after a successful login in LoginDialog.
+    /**
+     * Retrieves the last login password
+     */
```

Comment:
Some of the comments can be removed of getter setter here can be removed, unless they explain something not apparent from method name.